### PR TITLE
getAddressMap for pending ID

### PR DESCRIPTION
**Pull request by SanchayPahalwani**

### CODE REVIEW

File: src/infrastructure/implementations/blockstack-crux-user-repository.ts [hunk 1]
```diff
@@ -64,28 +65,41 @@ export class BlockstackCruxUserRepository implements ICruxUserRepository {
...
         return cruxUser;
     }
-    private getAddressMap = async (cruxId: CruxId, tag?: string): Promise<IAddressMapping> => {
+    private getAddressMap = async (cruxId: CruxId, tag?: string, ownerAddress?: string): Promise<IAddressMapping> => {
         const cruxPayFileName = CruxSpec.blockstack.getCruxPayFilename(cruxId);
-        const gaiaHub = await this.blockstackService.getGaiaHub(cruxId, tag);
-        const nameDetails = await this.blockstackService.getNameDetails(cruxId, tag);
-        if (!nameDetails.address) {
-            throw ErrorHelper.getPackageError(null, PackageErrorCode.MissingNameOwnerAddress, cruxId.toString());
+        const gaiaHub = (await this.blockstackService.getGaiaHub(cruxId, tag)) ? (await this.blockstackService.getGaiaHub(cruxId, tag)) : this.infrastructure.gaiaHub;
```
Comment (nandubatchu) [inline src/infrastructure/implementations/blockstack-crux-user-repository.ts]: avoid repetition of method calls here

File: src/infrastructure/implementations/blockstack-crux-user-repository.ts [hunk 2]
```diff
@@ -64,28 +65,41 @@ export class BlockstackCruxUserRepository implements ICruxUserRepository {
...
+        return new GaiaService(gaiaHub, this.cacheStorage).getContentFromGaiaHub(ownerAddress, cruxPayFileName);
     }
     private putAddressMap = async (addressMap: IAddressMapping, cruxDomainId: CruxDomainId, keyManager: IKeyManager): Promise<string> => {
         const cruxId = await this.blockstackService.getCruxIdWithKeyManager(keyManager, cruxDomainId);
         if (!cruxId) {
             throw ErrorHelper.getPackageError(null, PackageErrorCode.UserDoesNotExist);
         }
         const cruxPayFileName = CruxSpec.blockstack.getCruxPayFilename(cruxId);
-        const gaiaHub = (await this.blockstackService.isCruxIdAvailable(cruxId)) ? (await this.blockstackService.getGaiaHub(cruxId)) : this.infrastructure.gaiaHub;
+        const gaiaHub = (await this.blockstackService.getGaiaHub(cruxId)) ? (await this.blockstackService.getGaiaHub(cruxId)) : this.infrastructure.gaiaHub;
```
Comment (nandubatchu) [inline src/infrastructure/implementations/blockstack-crux-user-repository.ts]: same as above

File: src/infrastructure/implementations/blockstack-crux-user-repository.ts (hunk 1 above)
Comment (SanchayPahalwani) [reply src/infrastructure/implementations/blockstack-crux-user-repository.ts]: Done

File: src/infrastructure/implementations/blockstack-crux-user-repository.ts (hunk 2 above)
Comment (SanchayPahalwani) [reply src/infrastructure/implementations/blockstack-crux-user-repository.ts]: Done

**Pull request merged**